File: Tutorial3-2Scope2/src/main/java/com/example/tutorial3scope2/di/AppModule.java
package com.example.tutorial3scope2.di;

import android.content.Context;
import android.support.annotation.NonNull;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {
    Context appContext;

    public AppModule(@NonNull Context context) {
        this.appContext = context;
    }

    @Singleton
    @Provides
    Context provideAppContext() {
        return appContext;
    }
}

File: Tutorial5-4Dagger-Android-MultipleActivitiesAndFragments/src/main/java/com/smarttoolfactory/tutorial5_4dagger_android_multipleactivitiesandfragments/di/AppModule.kt
package com.smarttoolfactory.tutorial5_4dagger_android_multipleactivitiesandfragments.di

import android.app.Application
import android.content.Context
import android.content.SharedPreferences
import com.smarttoolfactory.tutorial5_4dagger_android_multipleactivitiesandfragments.main.MainActivitySubComponent
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

// The "subcomponents" attribute in the @Module annotation tells Dagger what
// Subcomponents are children of the Component this module is included in.

/**
 * AppModule of dagger-android with @ContributesAndroidInjector
 * does not declare sub-components inside @Module annotation
 *
 * @Module(subcomponents = {A.class, B.class}) is not used
 */
@Module(subcomponents = [MainActivitySubComponent::class])
abstract class AppModule {

    @Module
    companion object {

        @JvmStatic
        @Provides
        @Singleton
        fun provideSharedPreferences(application: Application): SharedPreferences {
            println("ApplicationModule context: $application")
            return application.getSharedPreferences("PrefName", Context.MODE_PRIVATE)
        }
    }

}